Feature: Schedule Appointment screen

twizerim opened the pull request:
### What does this pr do ?

This pr provides a screen that helps clients schedule appointment in any way according to their choice of method to be used in an appointment

#### Description of the task to be completed ?
The task will be completed, and the client will be able to use it to meet to a doctor in an easy way, according to his or her choice

#### How should this be manually tested?
 You can use : - npm run test
                         - npm run build

 And you reach my screen, work on  going to my appointment and you see upcoming , complete and canceled appointments. If you want to schedule appointment, you click on button, go to the reason for scheduling the appointment, go to the date and time, and then, by choice, go to cancel your appointment

and you see the icon indicating if you want to choose voice call , messaging or video call

![51_Light_my appointment upcoming-1](https://github.com/atlp-rwanda/commanders-rn-medica/assets/136312774/b6770233-6e27-4973-8ffd-da712df1f63d)

![Screenshot_1716304154](https://github.com/atlp-rwanda/commanders-rn-medica/assets/136312774/ef2944d3-d7fe-4dc0-bc7a-458e2d59a2b1)

![Screenshot_1716304184](https://github.com/atlp-rwanda/commanders-rn-medica/assets/136312774/7cf0f6e7-0b8b-4648-90c2-501231043700)

![Screenshot_1716304191](https://github.com/atlp-rwanda/commanders-rn-medica/assets/136312774/b7582436-2a72-4040-a4e7-79e148950397)

justinmihigo reviewed (changes requested):
Upgrade your expo to the newest version and do the npx expo install --fix to download compatible versions of other packages, also squash your commits

Commit:
debuging issues to implent true screen

rebasing dev branch

debuging issues to implent true screen

feat: doctor booking screens

fix issues caused by rebase with dev

fix svg issues on android and add primary color

fix: disable picking previous dates

Replace images with africans

fix: hours selection spacing, age field and problemem field

removing conflict

 removing duplicant incoplete code

adding new task called messaging

update newest version

correct icomplete screen

rebasing dev branch

Update README.md

Feature: Sign Up and Sign In

- Created SignUp screen with email and password input fields.
- Added SignIn screen with email and password input fields.
- Implemented active state for the input fields.

debuging issues to implent true screen

byaje

work on appointment screen in small divice

fix aproved feedback

appointment implemention

comment

voiceling

hbu

apucoming

to remove debuging

debug some errors

correct screen

tracking change

corect

correction of issues

complete appointment

appointment

work on comment and sugetions

button corect

boxshadow

remove asset alead found

rempve asset duplicant

Latiah commented:
Hey @twizerim  kindly improve the way you mentioned on how the pr  description "How should this be manually tested?
You can use : - npm run test
- npm run build"
- tell us how we will reach the screens you worked on instead and also include other screenshoots of the screens you have in this pr

justinmihigo commented:
@twizerim  resolve the conflicts

kibongos40 reviewed (changes requested):
Correct this typo on the **Yes, cancel** button

![image](https://github.com/atlp-rwanda/commanders-rn-medica/assets/54446750/757bfed3-1afe-411e-9618-7f0c16707952)

Latiah commented:
Awesome work @twizerim  but I did notice some parts on the screen where you did not use the font family like on the calender  and on some modals
On the modal the statement with discount it's font family is not Urbanist I guess on your implementation
![Screenshot_1716410493](https://github.com/atlp-rwanda/commanders-rn-medica/assets/99607224/078445ef-e274-4578-90c3-401316fd60f9)

![Screenshot_1716410901](https://github.com/atlp-rwanda/commanders-rn-medica/assets/99607224/a1f31e4a-f9c1-452d-bee5-f71d95ab99a3)
and the calender on the figma uses Mo, Tu ,... Su as the number of days you can change that very quickly

Latiah commented:
Also do remember to rebase

Adadie commented:
@twizerim how do I get to your screens?

Latiah commented:
> @twizerim how do I get to your screens?

from the appointment tab on the home screen that's where his screens are

twizerim commented:
> @twizerim how do I get to your screens?

you open the tab called appointment on home screen

Adadie commented:
> > @twizerim how do I get to your screens?
> 
> you open the tab called appointment on home screen

The tab isn't displaying anything on my end @twizerim

911samuel reviewed (approved):
great work bro!! things looks fine

justinmihigo reviewed (approved):
great work

chrysloi reviewed (changes requested):
Hi @twizerim you did a good but there are a few things you might have missed please look into them.
- The tabs look different with the design
- When canceling the appointment and you select 'Yes, Cancel' the modal doesn't close after redirecting.
- on the calendar the card seems to be having much space between them
below are images showing these parts

![Simulator Screen Shot - iPhone 14 Pro Max - 2024-05-27 at 16 00 46](https://github.com/atlp-rwanda/commanders-rn-medica/assets/99283140/987298e6-1122-4479-bc1b-fa16f8b79f43)
![Simulator Screen Shot - iPhone 14 Pro Max - 2024-05-27 at 16 01 00](https://github.com/atlp-rwanda/commanders-rn-medica/assets/99283140/aaf8218f-d62f-4e4b-96be-f39778c62048)
![Simulator Screen Shot - iPhone 14 Pro Max - 2024-05-27 at 16 01 43](https://github.com/atlp-rwanda/commanders-rn-medica/assets/99283140/59f288ea-bcf7-4529-9216-9347ff92a05a)

Adadie reviewed (changes requested):
@twizerim here is  the feedback I shared with you on the call, just for your reference:

1. The tabs for Ongoing, Completed and Cancelled should have a greyish color and the font-weight should be slightly higher.
2. The top margin at the Reschedule Appointment title is quite high.
3. Correct typos across all the screens
4. Remove the prefilled data in the text-area
5. Text in the text-area should be greyish
6. Reduce the spacing between the hour selection buttons

You may also refer to the screenshots below:
![Screenshot_20240527_175008_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/a3107692-469f-4e0e-8b25-895bb1947457)
![Screenshot_20240527_175008_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/a3107692-469f-4e0e-8b25-895bb1947457)
![Screenshot_20240527_174659_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/d2a88752-36c3-416f-8229-996f000ad4c6)
![Screenshot_20240527_174659_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/d2a88752-36c3-416f-8229-996f000ad4c6)
![Screenshot_20240527_174359_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/add30a14-4c94-488f-ba17-7bea571ae12c)
![Screenshot_20240527_174359_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/add30a14-4c94-488f-ba17-7bea571ae12c)
![Screenshot_20240527_175233_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/0ee867a8-e0db-4fe2-9a80-240e5629970c)
![Screenshot_20240527_175233_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/0ee867a8-e0db-4fe2-9a80-240e5629970c)
![Screenshot_20240527_175104_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/eea2d176-3807-46a6-96d4-e466c2d2cfda)
![Screenshot_20240527_175104_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/eea2d176-3807-46a6-96d4-e466c2d2cfda)
![Screenshot_20240527_175252_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/58fb5a37-6d15-4862-93b8-638d1c7aa3d7)
![Screenshot_20240527_175252_Expo Go](https://github.com/atlp-rwanda/commanders-rn-medica/assets/68109866/58fb5a37-6d15-4862-93b8-638d1c7aa3d7)

Adadie reviewed (changes requested):
Hey @twizerim awesome work. I like the improvements.
Would you please remove the assets that are already in the repo

Adadie commented on assets/appointmentIcon/first.png:
I believe you should already have this on dev

Adadie commented on assets/history.png:
This as well

Adadie commented on assets/home.png:
Also this

Adadie commented on assets/profile.png:
Also this is on dev

Adadie commented on assets/appointmentIcon/Group1.png:
Also this is on dev

Adadie commented on assets/appointmentIcon/Arrow.png:
Also this is on dev

Adadie commented on assets/Appoint.png:
Also this is on dev

Adadie commented on assets/Article.png:
Also this is on dev

twizerim replied:
assets/appointmentIcon/first.png  and  assets/appointmentIcon/Group1.png are not found in dev because they are required  in my task only and others can be removed
